span_marker/model_card_template.md

```diff
@@ -0,0 +1,174 @@
+# Run inference
+entities = model.predict("{{ predict_example | default("Amelia Earhart flew her single engine Lockheed Vega 5B across the Atlantic to Paris.", true)}}")
+```
+
+### Downstream Use
+You can finetune this model on your own dataset.
+
+<details><summary>Click to expand</summary>
+
+```python
+from span_marker import SpanMarkerModel, Trainer
+
+# Download from the {{ hf_emoji }} Hub
+model = SpanMarkerModel.from_pretrained("{{ model_id | default('span_marker_model_id', true) }}")
+
+# Specify a Dataset with "tokens" and "ner_tag" columns
+dataset = load_dataset("conll2003") # For example CoNLL2003
+
+# Initialize a Trainer using the pretrained model & dataset
+trainer = Trainer(
+    model=model,
+    train_dataset=dataset["train"],
+    eval_dataset=dataset["validation"],
+)
+trainer.train()
+trainer.save_model("{{ model_id | default('span_marker_model_id', true) }}-finetuned")
+```
+</details>
+
+<!--
+### Out-of-Scope Use
+
+*List how the model may foreseeably be misused and address what users ought not do with the model.*
+-->
+{% if tokenizer_warning %}
+### {{ warn_emoji }} Tokenizer Warning
+The [{{ encoder_name if encoder_name else encoder_id }}](https://huggingface.co/models/{{ encoder_id }}) tokenizer distinguishes between punctuation directly attached to a word and punctuation separated from a word by a space. For example, `Paris.` and `Paris .` are tokenized into different tokens. During training, this model is only exposed to the latter style, i.e. all words are separated by a space. Consequently, the model may perform worse when the inference text is in the former style.
+
+In short, it is recommended to preprocess your inference text such that all words and punctuation are separated by a space. Some potential approaches to convert regular text into this format are NLTK [`word_tokenize`](https://www.nltk.org/api/nltk.tokenize.word_tokenize.html) or spaCy [`Doc`](https://spacy.io/api/doc#iter) and join the resulting words with a space.
+{% endif %}
+<!--
+## Bias, Risks and Limitations
+
+*What are the known or foreseeable issues stemming from this model?*
+-->
+
+<!--
+### Recommendations
+
+*What are recommendations with respect to the foreseeable issues? For example, filtering explicit content.*
+-->
+
+## Training Details
+{% if train_set_metrics %}
+### Training Set Metrics
+{{ train_set_metrics }}{% endif %}{% if hyperparameters %}
+### Training Hyperparameters
+{% for name, value in hyperparameters.items() %}- {{ name }}: {{ value }}
+{% endfor %}{% endif %}{% if eval_lines %}
+### Training Results
+{{ eval_lines }}{% endif %}{% if co2_eq_emissions %}
+### Environmental Impact
+Carbon emissions were measured using [CodeCarbon](https://github.com/mlco2/codecarbon).
+- **Carbon Emitted**: {{ "%.3f"|format(co2_eq_emissions["emissions"] / 1000) }} kg of CO2
+- **Hours Used**: {{ co2_eq_emissions["hours_used"] }} hours
+
+### Training Hardware
+- **On Cloud**: {{ "Yes" if co2_eq_emissions["on_cloud"] else "No" }}
```

Comment:
Awesome!

Reply:
All from the `codecarbon` package, which is already integrated into `transformers`.